src/lib.rs

```diff
@@ -8,97 +11,99 @@ pub struct DB<T> {
 /// NB: (nota bene, or "take special note"): You will need to be explcit about the liftimes in this
 /// struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBView<T> {
-    entries: Vec<&T>,
+pub struct DBView<'a, T:'a> {
+    entries: Vec<&'a T>,
 }
 
 /// An mutably borrowed subset of a DB
 ///
 /// NB: You will need to be explcit about the liftimes in this struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBViewMut<T> {
-    entries: Vec<&mut T>,
+pub struct DBViewMut<'a, T:'a> {
+    entries: Vec<&'a mut T>,
 }
 
 /// Filters a DBView using the the given predicate.
-pub fn filter_one<T, F>(view: &DBView<T>, predicate: F) -> DBView< T>
+pub fn filter_one<'a, T, F>(view: &DBView<'a, T>, predicate: F) -> DBView<'a, T>
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+  DBView{ entries: view.entries.clone().into_iter().filter(|&x| predicate(x)).collect() }
```

Comment:
so this line could then be:
`DBView{ entries: view.entries.into_iter().filter(|&x| predicate(x)).cloned().collect() }` ?